### ISSUE TITLE
bug: run_sse silently drops runtime errors for non-native UI profiles

### ISSUE THREAD

**Issue by pumpkinsm** [zavora-ai/adk-rust#652]
## Description

When an agent stream returns an `Err(AdkError)`, the native AG-UI transport correctly emits a `RUN_ERROR` event.

The other `/api/run_sse` UI profiles, including the default ADK UI profile and legacy wrapped profiles such as A2UI, do not emit any corresponding error event. The stream simply returns.

Because the HTTP SSE response has already started with status 200, clients cannot use the HTTP status to distinguish this failure from a normal end-of-stream. They observe a silent EOF instead of the structured runtime error.

## Steps to Reproduce

Use an agent whose stream immediately returns an error:

```rust
struct ErrorStreamingTestAgent;

#[async_trait]
impl adk_core::Agent for ErrorStreamingTestAgent {
    async fn run(
        &self,
        _ctx: Arc<dyn adk_core::InvocationContext>,
    ) -> adk_core::Result<Pin<Box<dyn futures::Stream<Item = adk_core::Result<Event>> + Send>>> {
        Ok(Box::pin(stream::once(async {
            Err(adk_core::AdkError::model("provider unavailable"))
        })))
    }
}
```

Then send a request to `POST /api/run_sse` using either:

1. The default ADK UI profile.
2. A wrapped profile such as `"uiProtocol": "a2ui"`.

The HTTP response starts successfully, but the SSE body contains no runtime error event.

## Expected Behavior

All `/api/run_sse` profiles should deliver a structured error event when the underlying agent stream yields an error.

- Native AG-UI should continue emitting its protocol-native `RUN_ERROR`.
- ADK UI should receive an `Event` containing `error_code` and `error_message`.
- Wrapped profiles should receive the same runtime error through their existing protocol envelope.

## Actual Behavior

Only native AG-UI emits an error event. Other profiles terminate the stream without communicating the failure.

## Environment

- OS: macOS 15.7.2
- Rust version: 1.95.0
- ADK-Rust commit: f2edb385eaf24a2edcbd6b8230da53a5b365d4b1
- Affected crate(s): adk-server

## Deterministic SSE Experiment

A test agent was configured to return the following as its first and only stream item:

```rust
Err(adk_core::AdkError::model("provider unavailable"))
```

The complete HTTP response body was then collected for each UI profile.

| Profile | Upstream behavior | Fixed behavior |
|---|---|---|
| Default ADK UI | HTTP 200 followed by no SSE runtime error event | HTTP 200 with a structured event containing `error_code` and `provider unavailable` |
| Wrapped A2UI | HTTP 200 followed by no SSE runtime error event | HTTP 200 with an A2UI-wrapped event containing `ui_protocol: "a2ui"`, `error_code`, and `provider unavailable` |
| Native AG-UI | Emits protocol-native `RUN_ERROR` | Unchanged; continues emitting `RUN_ERROR` |

HTTP 200 is expected for an SSE response whose headers have already been committed. The error therefore needs to be represented inside the event stream.

This is a deterministic protocol-behavior test rather than a timing benchmark.

## Additional Context

The proposed fix converts the `AdkError` into the existing runtime `Event` error fields and passes it through the existing profile serializer.

It does not:

- Add or change a public endpoint.
- Change the successful event format.
- Change native AG-UI behavior.
- Introduce a new error envelope.
- Change HTTP status handling after the SSE response has started.

**Comment by jkmaina** [comment]
The fix in #653 passes code review and PR CI. Its remaining merge conflict is confined to adjacent changelog entries; both entries are preserved in the prepared resolution. This issue stays open until the fix merges.
